test/unit/logic/referral.spec.ts

```diff
@@ -677,6 +758,111 @@ describe('referral-component', () => {
         )
       })
     })
+
+    describe('when referral reaches 100 invited users', () => {
+      beforeEach(() => {
+        mockSlack.sendMessage.mockResolvedValueOnce(undefined)
+        mockReferralDb.findReferralProgress.mockResolvedValueOnce([
+          {
+            referrer: validReferrer,
+            invited_user: validInvitedUser,
+            status: ReferralProgressStatus.SIGNED_UP
+          }
+        ])
+        mockReferralDb.updateReferralProgress.mockResolvedValueOnce(undefined)
+        mockReferralDb.countAcceptedInvitesByReferrer.mockResolvedValueOnce(100)
+      })
+
+      it('should send Slack notification', async () => {
+        await referralComponent.finalizeReferral(validInvitedUser)
+
+        expect(mockSlack.sendMessage).toHaveBeenCalledWith({
```

Comment:
Would you mind using the functions that generate the message here to make the test shorter?